Archive/Imo/Imo2025Q3.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2025 Yi Yuan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yi Yuan
+-/
+import Mathlib.NumberTheory.LSeries.PrimesInAP
+import Mathlib.NumberTheory.LucasLehmer
+import Mathlib.NumberTheory.Multiplicity
+import Mathlib.Tactic.Simproc.Factors
+
+/-!
+# IMO 2025 Q3
+
+Let $\mathbb{N}+$ denote the set of positive integers. A function $f: \mathbb{N}+ \rightarrow
+\mathbb{N}+$ is said to be bonza if $f(a)$ divides $b^a-f(b)^{f(a)}$ for all positive integers $a$
+and $b$. Determine the smallest real constant $c$ such that $f(n) \leq c n$ for all bonza functions
+$f$ and all positive integers $n$.
+-/
+
+open Nat Int
+
+namespace Imo2025Q3
+
+/-- Define bonza functions -/
+def bonza : Set (ℕ → ℕ) :=
+  {f : ℕ → ℕ | (∀ a b : ℕ, 0 < a → 0 < b → (f a : ℤ) ∣ (b : ℤ) ^ a - (f b : ℤ) ^ (f a)) ∧
+    ∀ n, 0 < n → f n > 0}
```

Comment:
The normal mathlib convention would be to use `<` not `>`, so `0 < f n`. I don't think there's any good reason to deviate from that here (for example, the informal problem statement says "positive integers", not `> 0`, and personally I'd be inclined to use the `<` convention even for an informal statement that did say `>`). This applies in several places in this formalization.

Reply:
Also, consider defining a predicate `Bonza : (ℕ → ℕ) → Prop` rather than a set; if you do that, you can then put lemmas about bonza functions in a namespace `Bonza` and use dot notation when using those lemmas.